### ISSUE TITLE
Cursor blinking within tmux stops after upgrade kitty to 0.22

### ISSUE THREAD

**Issue by laggardkernel** [kovidgoyal/kitty#3906]
**Describe the bug**

Formerly, the cursor blinks normally within tmux. After upgrading kitty to 0.22.x. It doesn't blink within tmux anymore.

Envi

**To Reproduce**

Environments

- `kitty` with default config
- `tmux` 3.2a with following minimal config
- OS: macOS 10.14.6

```conf
# minimal tmux conf: test.conf
set-option -gs default-terminal "tmux-256color"
set-option -ga terminal-overrides ',*:Ss=\E[%p1%d q:Se=\E[5 q'
```

1. Open kitty with default config. 
2. Start tmux with `tmux -f ./test.conf -S ./test.socket`

Then the cursor stops blinking. Downgrading kitty back to 0.21. The problem disappears.

**Comment by kovidgoyal** [comment]
This is a bug in tmux that was masked by a bug in kitty. The bug in
kitty was fixed in 0.22.0 (see #3808)

Run

kitty --dump-commands tmux

In the list of commands tmux sends kitty you will see 
screen_reset_mode 12 1

This is the escape code

CSI ? 12 l

which is supposed to stop the cursor blinking, as described
here: https://invisible-island.net/xterm/ctlseqs/ctlseqs.html
(search the page for Ps = 1 2)

I dont use tmux so I cant comment on why it is doing this or how to get
it to stop doing it.

**Comment by laggardkernel** [comment]
@kovidgoyal Thanks for your reply. The clue you gave me did help.

To be frank, I know nothing about the source code of tmux. But searching `12l` gives no result in the source code. One of the possible reason is that `tmux` is firing `cnorm`. 

```
❯ infocmp xterm-kitty | grep '12l'
        clear=\E[H\E[2J, cnorm=\E[?12l\E[?25h, cr=^M,
```

I tried to override `cnorm` within tmux and my cursor blinks now.

```
# tmux.conf
# change \E[?12l to \E[?12h
set-option -ga terminal-overrides ',xterm-kitty:cnorm=\E[?12h\E[?25h'
```

## Ref

Clues lead me to the current solution

- [Cursor blinking discrepancy on Linux](https://github.com/tmux/tmux/issues/2682) mentions `cnorm` and related change
- [Fix so tmux correctly sends the cvvis (cursor very visible) capability](https://github.com/tmux/tmux/commit/e8224fb0d123ca9042b0703a35e1a33f0787d327), possible code why tmux send `cnorm`
- [Change cursor shape by modifying `cnorm` in terminfo](https://unix.stackexchange.com/a/629826)
